app/forms/fields/year_date_field.py

```diff
@@ -26,11 +28,28 @@ def data(self):
 
 
 class YearDateField(FormField):
-    def __init__(self, validators, **kwargs):
+    def __init__(
+        self,
+        *,
+        validators: Sequence[Callable],
+        label: str | None = None,
+        description: str,
+        **kwargs: Any,
+    ):
         form_class = get_form_class(validators)
-        super().__init__(form_class, **kwargs)
-
-    def process(self, formdata, data=None, extra_filters=None):
+        super().__init__(
+            form_class,
+            label=label,
+            description=description,
```

Comment:
How did you decide which kwargs to pull out of the dictionary and type over which ones to leave? just wondering since it differs between them

Reply:
Since we haven't removed `**kwargs` I don't think it is worth pulling these out, the kwargs keep the field extensible. If we were typing each one, then the kwargs don't provide any value. I would learn towards keeping the field generic and just using kwargs. See what others think.

Reply:
I would lean that way as well, unless specifically validating one/ its type creates issues, I think it's fine staying as kwargs

Reply:
Thanks both 👍  Just to confirm after a quick team discussion, we've gone with the preferred way of keeping `kwargs`  for caller flexibility with the third party library, and only extracting a `kwarg` if the scope of the class calls for it. I think I've updated it in places where this should be the case now?